soundfile.py

```diff
@@ -167,6 +167,38 @@
     __libsndfile_version__ = __libsndfile_version__[len('libsndfile-'):]
 
 
+class SoundFileError(Exception):
+    """Base class for all SoundFile-specific errors."""
+    pass
+
+class SoundFileRuntimeError(SoundFileError, RuntimeError):
+    """SoundFile runtime error.
+
+    Errors that used to be `RuntimeError`."""
+    pass
+
+class LibsndfileError(SoundFileRuntimeError):
+    """libsndfile errors.
+
+
+    Attributes
+    ----------
+    code
+        libsndfile internal error number
+
+    error_string
+        raw libsndfile error message
+    """
+    def __init__(self, code, prefix=""):
+        SoundFileRuntimeError.__init__(self, code, prefix)
+        self.code = code
+        self.prefix = prefix
+        err_str = _snd.sf_error_number(code)
```

Comment:
What about doing this in `__str__`?

I normally try to do as much of the work as late as possible.

Is there a reason to do this eagerly?

Now that I'm thinking about it, we could even make the `error_string` a `@property`. I've never done such a thing in an exception class, and I don't know whether that's a good idea ...